tools/wptrunner/wptrunner/executors/base.py

```diff
@@ -307,14 +307,18 @@ def run_test(self, test):
 
         self.runner.send_message("test_ended", test, result)
 
-    def server_url(self, protocol):
+    def server_url(self, protocol, subdomain=None):
         scheme = "https" if protocol == "h2" else protocol
-        return "%s://%s:%s" % (scheme,
-                               self.server_config["browser_host"],
-                               self.server_config["ports"][protocol][0])
+        host = self.server_config["browser_host"]
+        if subdomain:
+            # The only supported subdomain filename flag is "www".
```

Comment:
Nit; can we assert this? Something like:

```suggestion
            # The only supported subdomain filename flag is "www".
            assert subdomain == "www", "Unsupported subdomain {}".format(subdomain)
```

Reply:
Oh right, I just realized that `subdomain` is a bool, so we can't assert anything useful here. This seemed like the easiest approach for a single supported subdomain flag name -- if we want to support `www1.www2`, etc, then we could store the actual subdomain value(s).

Reply:
Ah, ok. In which case should the default be `False` rather than `None` above?

Reply:
Aaaand, you're right. Thanks :)